### ISSUE TITLE
Add optional argument to dotnet new for adding .gitignore

### ISSUE THREAD

**Issue by isaacrlevin** [dotnet/sdk#28011]
### Is your feature request related to a problem? Please describe.

Based off of Twitter thread for folks
https://twitter.com/isaacrlevin/status/1571908208675147776

### Describe the solution you'd like
Adding a new argument to `dotnet new` that will allow adding a `.gitignore` to the output of the command. For instance
```bash
dotnet new web --add-git
```

**Comment by baronfel** [comment]
First step would be to identify the templates that _should_ have this option. Only some of them are in this repo. You can check the README on the dotnet/templating repo for a more up-to-date list of repos that own specific templates - I try to keep that up to date.  Then, for each project template that should have this option, you'd add a new `symbol` to the [template.json](https://github.com/dotnet/sdk/blob/main/template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ConsoleApplication-CSharp/.template.config/template.json#L21) - note that there are different templates for each supported language, so you'd need to replicate that symbol.

Once you have a symbol defined, you'd add a [postAction](https://github.com/dotnet/sdk/blob/main/template_feed/Microsoft.DotNet.Common.ProjectTemplates.7.0/content/ConsoleApplication-CSharp/.template.config/template.json#L125) that has a `condition` that checks the symbol you defined earlier. I'd suggest looking at the `run script` [postAction](https://github.com/dotnet/templating/wiki/Post-Action-Registry#run-script). Once you've made the changes, you can test by running `dotnet new install path/to/template/folder` to install the templates in that folder, invoke them, then uninstall with `dotnet new uninstall path/to/template/folder`.

**Comment by isaacrlevin** [comment]
Got it, so there isn't a universal template that all the other templates inherit from? I was thinking at the same level the output argument, but you are saying it needs to be done at every template.

**Comment by baronfel** [comment]
Correct - there's no inheritance here. Each template is an island.

**Comment by isaacrlevin** [comment]
Got it. So this change will need to be in a few different repos than. I will take a look

**Comment by vlada-shubina** [comment]
this is very much related to https://github.com/dotnet/templating/issues/3208 and https://github.com/dotnet/templating/issues/2771.

**Comment by isaacrlevin** [comment]
@vlada-shubina is there work being done in this space? I would hate to start the process of updating every template if there is a new process coming at some point.

**Comment by vlada-shubina** [comment]
No, unfortunately there is no definite timeline for those features at the moment.

**Comment by vlada-shubina** [comment]
We decided it falls in scope of https://github.com/dotnet/templating/issues/3208 or https://github.com/dotnet/templating/issues/2771, but there is no definite timeline on that yet.
